src/FACTFinder/Core/AbstractEncodingConverter.php

```diff
@@ -58,12 +58,19 @@ function __construct(
      */
     protected function convert($inCharset, $outCharset, $data)
     {
-        if (FF::isInstanceOf($data, 'Util\Parameters'))
+        if ($data instanceof Parameters)
         {
-            $result = FF::getInstance(
-                'Util\Parameters',
-                $this->convert($inCharset, $outCharset, $data->getArray())
-            );
+            if (count($data->getArray()) == 1 && current(array_keys($data->getArray())) == '')
```

Comment:
Was it a bug with empty parameter names or with *empty* (0-sized) arrays? I don't get the cause for the bug by seeing this solution; it just confuses me, as a reader of this code.